pennylane/pulse/transmon.py

```diff
@@ -215,3 +222,264 @@ def __add__(self, other):
             return TransmonSettings(new_connections, new_omega, new_g, anharmonicity=new_anh)
 
         return self
+
+
+def transmon_drive(amplitude, phase, freq, wires, d=2):
+    r"""Returns a :class:`ParametrizedHamiltonian` representing the drive term of a transmon qubit.
+
+    The Hamiltonian is given by
+
+    .. math::
+
+        \Omega(t) \left(e^{i (\phi(t) + \nu t)} a_q + e^{-i (\phi(t) + \nu t)} a^\dagger_q\right)
```

Comment:
This is technically not the formulation we use in the code, so when you print a transmon drive, you see it in terms of sigma_x and sigma_y, i.e.

```
>>> H(params, t)
(0.1*(PauliX(wires=[0]))) + (0.0*(-1*(PauliY(wires=[0]))))
```

Should this be reformulated to be consistent with the function output? Or do we not care so much as long as its mathematically equivalent?

Reply:
Same question about the `drive` function I guess, though not in this PR

Reply:
Definitely something to get back to when we introduce qudit support. For the moment I see little point in it since there is no creation and annihilation operator in PennyLane and it always comes down to decomposition in terms of $\sigma^x$ and $\sigma^y$. I tried showing this in the example avoid confusion:
![image](https://user-images.githubusercontent.com/43949391/233303888-b2324979-8747-494b-b0a9-d3b695021929.png)